src/libraries/Microsoft.Extensions.DependencyInjection.Specification.Tests/src/ActivatorUtilitiesTests.cs

```diff
@@ -214,9 +221,9 @@ public void TypeActivatorCreateFactoryDoesNotAllowForAmbiguousConstructorMatches
         }
 
         [Theory]
-        [InlineData("", "string")]
-        [InlineData(5, "IFakeService, int")]
-        public void TypeActivatorCreateInstanceUsesFirstMathchedConstructor(object value, string ctor)
+        [InlineData("", "IFakeService, string", "string")]
+        [InlineData(5, "IFakeService, int", "IFakeService, int")]
+        public void TypeActivatorCreateInstanceUsesLongestAvailableConstructor(object value, string ctor, string oldBehavior)
```

Comment:
`oldBehavior` is not a good name to choose. "Old" as relative to what? It quickly becomes ambiguous.